jembi/platform
# PLAT-608 / PLAT-633 / PLAT-634 : keycloak package

## Motivation 
This PR includes the following : 
- Add logic for clustered mode : https://jembiprojects.jira.com/browse/PLAT-608
- Add Postgres to store Keycloak Realm/User data : https://jembiprojects.jira.com/browse/PLAT-633
- Add Realm provisioning for grafana client : https://jembiprojects.jira.com/browse/PLAT-634
- Fixes a promotheus issue
- Refactor all configs into env vars and expose the client id & secret

## How to test 
1. Fetch the branch 
2. Build the platform image
3. `./platform-linux init identity-access-manager-keycloak --only --dev --env-file=.env.local`
4. `./platform-linux init monitoring --only --dev --env-file=.env.local`
5. Access grafana `http://localhost:3000` and click on "Login with KeyCloak"
6. Login with KeyCloak using the following credentials : username = test / password = dev_password_only
7. Should be logged to grafana as admin.

[PLAT-608]: https://jembiprojects.jira.com/browse/PLAT-608?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PLAT-633]: https://jembiprojects.jira.com/browse/PLAT-633?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PLAT-634]: https://jembiprojects.jira.com/browse/PLAT-634?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
